### PR TITLE
Support variables in product name

### PR DESCRIPTION
**Pull request by dcvz**
### Short description 📝

Users sometimes want to have different app display names for different schemes. It can be useful to distinguish builds if they both live on one device.

This PR adds the ability to use variables in `productName`.

### Solution 📦

Replace all the variables in the product name and run the validation of the character on the remaining string. Inspired by https://github.com/tuist/tuist/pull/252.

### Concerns 🤔

I find it weird that you can define `settings` per scheme but not some other things that you would also want to customize i.e. `productName` and `bundleId`.

I also thought that adding the `PRODUCT_NAME` setting to the `debug` and `release` scheme would work.. but it didn't and that was a not intuitive.

I realize that this is probably because they are necessary to generate a project. And because they're already declared in the target, it doesn't make sense to then take the `settings` into account too -- but it leads to this "workaround" of having to customize them via variables.

_NOTE: I'm relatively new to this project. I attempted to search the repository to see if this had been asked or if it was possible somehow, but didn't find an answer. Please let me know if I've overlooked something._

### CODE REVIEW

**Comment by dcvz** [comment]
Hmm, not exactly sure why there's indention issues :/

File: Sources/TuistGenerator/Linter/TargetLinter.swift [hunk 1]
```diff
@@ -61,9 +61,15 @@ class TargetLinter: TargetLinting {
     }
 
     private func lintProductName(target: Target) -> [LintingIssue] {
+        var productName = target.productName
+
+        // Remove any interpolated variables
+        productName = productName.replacingOccurrences(of: "\\$\\{.+\\}", with: "", options: .regularExpression)
+        productName = productName.replacingOccurrences(of: "\\$\\(.+\\)", with: "", options: .regularExpression)
+
         let allowed = CharacterSet(charactersIn: "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_")
```
Comment (pepibumur) [inline Sources/TuistGenerator/Linter/TargetLinter.swift]: What about including the `$`, `{` and `}` characters in this list instead?

Comment (dcvz) [reply Sources/TuistGenerator/Linter/TargetLinter.swift]: I thought about it, but as mentioned I followed what you did in #252 so thought you'd prefer it that way :P

Comment (pepibumur) [reply Sources/TuistGenerator/Linter/TargetLinter.swift]: 😂 there must be a reason for doing it that way but I can't remember it.

**Pull request merged**